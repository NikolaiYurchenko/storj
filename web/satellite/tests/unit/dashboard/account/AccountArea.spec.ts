// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { mount, shallowMount } from '@vue/test-utils';
import AccountArea from '@/components/account/AccountArea.vue';
import * as sinon from 'sinon';

describe('AccountArea.vue', () => {
    it('renders correctly', () => {

        const wrapper = shallowMount(AccountArea, {
            mocks: {
                $router: {
                    push: () => {}
                }
            },
            stubs: ['router-view', 'router-link']
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('trigger redirection on mounted correctly', () => {
        let routerSpy = sinon.spy();

        const wrapper = mount(AccountArea, {
            mocks: {
                $router: {
                    push: routerSpy
                }
            },
            stubs: ['router-view', 'router-link']
        });

        expect(routerSpy.callCount).toBe(1);
    });
});
