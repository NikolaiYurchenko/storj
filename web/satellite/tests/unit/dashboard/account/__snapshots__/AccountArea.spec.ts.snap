// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountArea.vue renders correctly 1`] = `
<div class="account-area-container">
  <h1>Account</h1>
  <tabnavigation-stub navigation="[object Object]" class="account-area-container__navigation"></tabnavigation-stub>
  <router-view-stub></router-view-stub>
</div>
`;
